samples/gen-js/index.js

```diff
@@ -0,0 +1,368 @@
+
+const discovery = require("@clever/discovery");
+const request = require("request");
+const url = require("url");
+const opentracing = require("opentracing");
+
+// go-swagger treats handles/expects arrays in the query string to be a string of comma joined values
+// so...do that thing. It's worth noting that this has lots of issues ("what if my values have commas in them?")
+// but that's an issue with go-swagger
+function serializeQueryString(data) {
+  if (Array.isArray(data)) {
+    return data.join(",");
+  }
+  return data;
+}
+
+module.exports = class SwaggerTest {
+
+  address;
+  timeout;
+
+  constructor(options) {
+    options = options || {};
+
+    if (options.discovery) {
+      this.address = discovery("swagger-test", "default").url();
+    } else if (options.address) {
+      this.address = options.address;
+    } else {
+      throw new Error("Cannot initialize swagger-test without discovery or address");
+    }
+    if (options.timeout) {
+      this.timeout = options.timeout
+    }
+  }
+
+  getBooks = (authors, available, state, published, snakeCase, completed, maxPages, minPages, pagesToTime, options, cb) => {
```

Comment:
I'm really not a fan of this... how hard would it be to take in an object here, attempt to pull these values out as keys, and fail if there are extraneous keys (i.e. a super light-weight validation that might eventually be full-on json-schema validation)?